itis.playground/Pages/Game.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,238 @@
+import Foundation
+
+enum Fraction: String{
+    case water = "Water"
+    case fire = "Fire"
+    case earth = "Earth"
+    case air = "Air"
+}
+
+protocol Student{
+    var fraction: Fraction { get set }
+    
+    var isAlive: Bool { get set }
+    var HP: Int { get set }
+    var damage: Int { get set }
+    var cry: String { get set }
+    var name: String { get set }
+    init(name: String)
+}
+
+protocol Advantage {
+    var advantage: Int { get }
+    func getAdv() -> Void
+}
+
+protocol Disadvantage {
+    var disadvantage: Int { get }
+    func getDis() -> Void
+}
+
+class Arena {
+    
+    var firstTeam: Team
+    
+    var secondTeam: Team
+    
+    init(team1: Team, team2: Team) {
+        firstTeam = team1
+        secondTeam = team2
+    }
+    
+    func startBattle() {
+        while firstTeam.isAlive, secondTeam.isAlive {
+            print(firstTeam.students[0]?.cry ?? "")
+            secondTeam.getDamage(damage: firstTeam.fullDamage())
+            if secondTeam.isAlive {
+                print(secondTeam.students[0]?.cry ?? "")
+                firstTeam.getDamage(damage: secondTeam.fullDamage())
+            }
+        }
+    }
+}
+
+class Team {
+    var students: [Optional<Student>]
```

Comment:
```suggestion
    var students: [Student?]
```
лучше так

Reply:
но в массивах это немного странно использовать 
лучше уж пустой массив
```suggestion
    var students: [Student]
```